src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -811,6 +811,10 @@ internal static void ValidateFunction(FunctionDescriptor function, Dictionary<st
 
                 httpFunctions.Add(function.Name, httpTrigger);
             }
+            if (environment != null && environment.IsFlexConsumptionSku() && function.Metadata.IsLegacyBlobTriggerFunction())
+            {
+                throw new InvalidOperationException($"Regular blob trigger for function '{function.Name}' is not supported in Flex Consumption plan. Please change the blob trigger to use Event Grid.");
```

Comment:
Should we also consider emitting a diagnostic error in this case?

Reply:
I followed the pattern we have for other error cases; I do not see "emitting a diagnostic errors" there:
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L802
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L829